hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieCreateHandle.java

```diff
@@ -173,6 +179,26 @@ record = record.prependMetaFields(schema, writeSchemaWithMetaFields, new Metadat
     }
   }
 
+  private void trackMetadataIndexStats(HoodieRecord record) {
```

Comment:
move all the index stats related changes into its corresbonding utility classes.

Reply:
Discussion in https://github.com/apache/hudi/pull/13449#discussion_r2157023760